### ISSUE TITLE
add QhullError to the public API for scipy.spatial?

### ISSUE THREAD

**Issue by grlee77** [scipy/scipy#14999]
A recent PR to scikit-image (not yet merged) involves use of `QhullError` as imported from `scipy.spatial.qhull`, but it looks like this is not public API (reference discussion at https://github.com/scikit-image/scikit-image/pull/6008/files#r744350997). My question is whether this attribute should be public? The use case is to check for this error type and return a sensible result when a convex hull for an object cannot be found.

It seems this would just involve adding it to `__all__` [here](https://github.com/scipy/scipy/blob/7c904a77ad3d239bc35f18222810c9454e5441c3/scipy/spatial/_qhull.pyx#L31) so that it can be imported using `from scipy.spatial import QhullError`.

ping @czgdp1807 and @rgommers who worked on defining the `scipy.spatial` API recently in #14923

**Comment by tylerjereddy** [comment]
Not sure what Ralf thinks, but most of the work under the hood is done by Qhull so at first glance it doesn't seem like a bad idea to make the error object public/"official." 

Handling degenerate conditions/lack of general position can also be a pain for many of the diagrams so if we can leave that to Qhull as much as possible that might be a win (and folks would presumably want to be able to catch those errors sometimes). Maybe there are downsides I'm not thinking of though.

**Comment by tirthasheshpatel** [comment]
> My question is whether this attribute should be public? The use case is to check for this error type and return a sensible result when a convex hull for an object cannot be found.

When I integrated the UNU.RAN library, I made a decision to make the error object public for the same reason @grlee77 mentioned. Not sure if it applies to Qhull but generally having the error object public also improves the readability of the user code (e.g. in a try...except block, instead of `except RuntimeError`, we could have `except QhullError` which is much more informative).
